File: src/main/java/com/kasisoft/libs/common/types/GeoLocation.java
package com.kasisoft.libs.common.types;

/**
 * @author daniel.kasmeroglu@kasisoft.com
 */
public record GeoLocation(double latitude, double longitude) {

    public boolean isValid() {
        return !(Double.isNaN(latitude) || Double.isNaN(longitude));
    }

} /* ENDRECORD */

File: src/main/kotlin/com/kasisoft/libs/common/types/GeoLocation.kt
package com.kasisoft.libs.common.types

/**
 * @author daniel.kasmeroglu@kasisoft.com
 */
data class GeoLocation(var latitude: Double, var longitude: Double) {

    fun isValid(): Boolean = !(latitude.isNaN() || longitude.isNaN())

} /* ENDCLASS */
